src/dataflow/src/render/context.rs

```diff
@@ -490,6 +490,13 @@ where
     ) {
         mfp.optimize();
         let mfp_plan = mfp.into_plan().unwrap();
+        // This path results in a slightly smaller and faster
+        // dataflow graph, and is intended to fix
+        // https://github.com/MaterializeInc/materialize/issues/10507
+        if mfp_plan.is_identity() && !matches!(key_val, Some((_, Some(_)))) {
```

Comment:
I can't read the `matches!` macro and generally prefer to avoid it. Like, no clue what the test is doing. Maybe put more words in the comment to avoid having to read `as_specific_collection` and the doccomments for `key_val` to understand what is happening.